configure.ac

```diff
@@ -3262,15 +3262,21 @@ dnl
 dnl Simple Generational checks (sgen)
 dnl
 SGEN_DEFINES=
-AC_ARG_WITH(sgen, [  --with-sgen=yes,no             Extra Generational GC, default=yes],[buildsgen=$with_sgen],[buildsgen=yes])
+AC_ARG_WITH(sgen,                    [  --with-sgen=yes,no             Extra Generational GC, default=yes],[buildsgen=$with_sgen],[buildsgen=yes])
+AC_ARG_WITH(sgen-default-concurrent, [  --with-sgen-default-concurrent=yes,no             Use Concurrent GC, default=no],[],[with_sgen_default_concurrent=no])
 if test x$buildsgen = xyes; then
    AC_DEFINE(HAVE_MOVING_COLLECTOR, 1, [Moving collector])
-   SGEN_DEFINES="-DHAVE_SGEN_GC"
    if test "x$gc_msg" = "x"; then
       gc_msg="sgen"
    else
       gc_msg="sgen and $gc_msg"
    fi
+
+   if test x$with_sgen_default_concurrent = xno; then
+       SGEN_DEFINES="-DHAVE_SGEN_GC"
+   else
```

Comment:
We should use AC_DEFINE for this define as it doesn't interfere with boehm.

Additionally, it should be namespaced.